File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/hardware/physical/requests/AccelerometerRequest.java
package org.firstinspires.ftc.teamcode.utils.hardware.physical.requests;

import com.michaell.looping.ScriptParameters;
import com.qualcomm.robotcore.hardware.AccelerationSensor;
import com.qualcomm.robotcore.hardware.HardwareMap;
import org.firstinspires.ftc.teamcode.utils.hardware.physical.accessors.AccelerometerGlobalAccess;
import org.firstinspires.ftc.teamcode.utils.hardware.physical.data.AccelerometerData;

public class AccelerometerRequest extends ScriptParameters.Request {

    private final AccelerationSensor SENSOR;

    public AccelerometerRequest(String name, HardwareMap hardwareMap) {
        super(name);
        SENSOR = hardwareMap.get(AccelerationSensor.class, name);
        SENSOR.resetDeviceConfigurationForOpMode();
    }

    @Override
    public Object issueRequest(Object o) {
        return new AccelerometerData(SENSOR.getAcceleration(), SENSOR.status());
    }

    @Override
    public Class getOutputType() {
        return AccelerometerData.class;
    }

    @Override
    public Class getInputType() {
        return Object.class;
    }

}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utils/hardware/requests/AccelerometerRequest.kt
package org.firstinspires.ftc.teamcode.utils.hardware.requests

import com.michaell.looping.ScriptParameters
import com.qualcomm.robotcore.hardware.AccelerationSensor
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.utils.hardware.data.AccelerometerData

class AccelerometerRequest(name: String, hardwareMap: HardwareMap) : ScriptParameters.Request(name) {
    val accelerometer: AccelerationSensor

    init {
        accelerometer = hardwareMap.get(AccelerationSensor::class.java, name)
        accelerometer.resetDeviceConfigurationForOpMode()
    }

    override fun issueRequest(o: Any): Any {
        return AccelerometerData(
            accelerometer.acceleration,
            accelerometer.status()
        )
    }

    override fun getOutputType(): Class<*> {
        return AccelerometerData::class.java
    }

    override fun getInputType(): Class<*> {
        return Any::class.java
    }
}